aten/src/ATen/native/cuda/UpSampleBilinear2d.cu

```diff
@@ -1,25 +1,312 @@
+
+        auto idata = input.packed_accessor<scalar_t, 4>();
+        auto odata = output.packed_accessor<scalar_t, 4>();
+
+        const accscalar_t rheight = area_pixel_compute_scale<accscalar_t>(
+            input_height, output_height, align_corners);
+        const accscalar_t rwidth = area_pixel_compute_scale<accscalar_t>(
+            input_width, output_width, align_corners);
+
+        upsample_bilinear2d_out_frame<scalar_t, accscalar_t>
+            <<<(num_kernels + num_threads - 1) / num_threads,
+               num_threads,
+               0,
+               stream>>>(
+                num_kernels, rheight, rwidth, align_corners, idata, odata);
+      });
+
+      AT_CUDA_CHECK(cudaGetLastError());
+}
+
+static void upsample_bilinear2d_backward_out_cuda_template(
+    Tensor& grad_input,
+    const Tensor& grad_output_,
+    IntArrayRef output_size,
+    IntArrayRef input_size,
+    bool align_corners) {
+  TensorArg grad_input_arg{grad_input, "grad_input", 1},
+      grad_output_arg{grad_output_, "grad_output_", 2};
+  checkAllSameGPU(
+      "upsample_bilinear2d_backward_out_cuda",
+      {grad_output_arg, grad_input_arg});
+
+  AT_CHECK(
+      output_size.size() == 2,
+      "It is expected output_size equals to 2, but got size ",
+      output_size.size());
+
+  AT_CHECK(
+      input_size.size() == 4,
+      "It is expected input_size equals to 4, but got size ",
+      input_size.size());
+
+  int64_t output_height = output_size[0];
+  int64_t output_width = output_size[1];
+
+  int64_t nbatch = input_size[0];
+  int64_t channels = input_size[1];
+  int64_t input_height = input_size[2];
+  int64_t input_width = input_size[3];
+
+  upsample_2d_shape_check(
+      Tensor(),
+      grad_output_,
+      nbatch,
+      channels,
+      input_height,
+      input_width,
+      output_height,
+      output_width);
+
+  Tensor grad_output = grad_output_.contiguous();
+
+  grad_input.resize_({nbatch, channels, input_height, input_width});
+  grad_input.zero_();
+
+  const int64_t num_kernels = output_height * output_width;
+  const int64_t num_threads =
+      at::cuda::getCurrentDeviceProperties()->maxThreadsPerBlock;
+  cudaStream_t stream = at::cuda::getCurrentCUDAStream();
+
+  AT_DISPATCH_FLOATING_TYPES_AND_HALF(
+      grad_output.scalar_type(), "upsample_bilinear2d_backward_out_frame", [&] {
+        using accscalar_t = at::acc_type<scalar_t, true>;
+
+        auto idata = grad_input.packed_accessor<scalar_t, 4>();
+        auto odata = grad_output.packed_accessor<scalar_t, 4>();
+
+        const accscalar_t rheight = area_pixel_compute_scale<accscalar_t>(
+            input_height, output_height, align_corners);
+        const accscalar_t rwidth = area_pixel_compute_scale<accscalar_t>(
+            input_width, output_width, align_corners);
+
+        upsample_bilinear2d_backward_out_frame<scalar_t, accscalar_t>
+            <<<(num_kernels + num_threads - 1) / num_threads,
```

Comment:
Arguably we should just add a `ceil_div` function and use it, it's a lot clearer. Doesn't have to be this PR.

Reply:
sounds good. do you have any idea what file this function should be?

Reply:
Not really. I guess we could always dump it in `Utils.h` for a start, perhaps in the at::impl namespace so that users don't start using it.

Reply:
sounds good! :) thanks for the feedback!

Reply:
https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/cuda/CUDAApplyUtils.cuh#L615-L618

Reply:
awesome! thanks @ngimel I will change to that!